intyg-proxy-service/integration-hsa/src/main/java/se/inera/intyg/intygproxyservice/integration/organization/client/converter/GetUnitResponseTypeConverter.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (C) 2023 Inera AB (http://www.inera.se)
+ *
+ * This file is part of sklintyg (https://github.com/sklintyg).
+ *
+ * sklintyg is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * sklintyg is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package se.inera.intyg.intygproxyservice.integration.organization.client.converter;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import se.inera.intyg.intygproxyservice.integration.api.organization.model.Unit;
+import se.riv.infrastructure.directory.organization.getunitresponder.v4.GetUnitResponseType;
+
+@Service
+@RequiredArgsConstructor
+public class GetUnitResponseTypeConverter {
+
+  private final UnitTypeConverter unitTypeConverter;
+
+  public Unit convert(GetUnitResponseType type) {
+    if (type == null || type.getUnit() == null) {
```

Comment:
If the unit is not found, might be a good idea to log this event, to keep existing functionality. Maybe something like:

```java
log.warn("No unit was found with hsaId '{}', returning empty unit", getUnitRequestDTO.getHsaId());
```

Reply:
I haven't added any logging for the other implementations when receiving null. Is this function the only one that has it or should it maybe not be placed in IPS?

Reply:
Separated jira is created